internal/provider/vaultsecrets/vault_secrets_utils.go

```diff
@@ -16,8 +16,28 @@ import (
 	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
 )
 
-// sharedIntegrationAttributes are the attributes shared between all the Vault Secrets integrations
-var sharedIntegrationAttributes = map[string]schema.Attribute{
+type Provider string
+
+const (
+	ProviderAWS Provider = "aws"
+	ProviderGCP Provider = "gcp"
+)
```

Comment:
As a follow-up we could publish the full list of providers in our public protos (like we do for capabilities) instead of redefining it here. I'm not sure we want to publish a static list with the ecosystem initiative though.

Reply:
Ya I could see that being useful. HCP CLI repo has also defined these similar to here. Also unsure if that's desired with the ecosystem initiative.

Reply:
I'll defer for now.